upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -149,27 +155,47 @@ func (e *Elastigroup) Find(c *fi.Context) (*Elastigroup, error) {
 
 		// Image.
 		{
-			image, err := resolveImage(cloud, fi.StringValue(lc.ImageID))
-			if err != nil {
-				return nil, err
+			if e.ImageID != nil && lc.ImageID != nil &&
+				fi.StringValue(lc.ImageID) != fi.StringValue(e.ImageID) {
+				image, err := resolveImage(cloud, fi.StringValue(e.ImageID))
+				if err != nil {
+					return nil, err
+				}
+				if fi.StringValue(image.ImageId) == fi.StringValue(lc.ImageID) {
+					actual.ImageID = e.ImageID
+				}
 			}
-			actual.ImageID = image.Name
 		}
 
 		// Tags.
 		{
 			if len(lc.Tags) > 0 {
 				actual.Tags = make(map[string]string)
 				for _, tag := range lc.Tags {
-					actual.Tags[*tag.Key] = *tag.Value
+					actual.Tags[fi.StringValue(tag.Key)] = fi.StringValue(tag.Value)
 				}
 			}
 		}
 
 		// Security groups.
 		{
-			for _, sg := range lc.SecurityGroupIDs {
-				actual.SecurityGroups = append(actual.SecurityGroups, &awstasks.SecurityGroup{ID: fi.String(sg)})
+			for _, sg := range e.SecurityGroups {
+				t := &awstasks.SecurityGroup{ID: sg.ID}
+				if sg.VPC != nil {
+					t.VPC = &awstasks.VPC{ID: sg.VPC.ID}
+				}
+				s, err := t.Find(c)
+				if err != nil {
+					return nil, err
+				}
+				if s != nil {
+					for _, sgID := range lc.SecurityGroupIDs {
+						if fi.StringValue(s.ID) == sgID {
+							actual.SecurityGroups = append(actual.SecurityGroups, s)
```

Comment:
It looks like we're driving this from the expected values, and ignoring other security groups we find on the LaunchConfiguration.  Not sure what we're doing here, and I'm not sure it's right.  If it is right it does need a comment because it's unusual..